### ISSUE TITLE
Fixsim.js project

### ISSUE THREAD

**Issue by ka-sh** [hongkongjs/ideas#1]
FIX simulator is a  sell-side FIX trading simulator to help anyone who is dealing with FIX protocol to fully test his "EMS" (Execution management system) and have the ability to simulate any scenario required to be FIX certified.

Currently there is very few inconvenient options that either need to pay to use, or you have to manually go through the test cases and not all of them supported.

Why? 
Sure banks, and private investment banks are all evil but on the other hand number of Fintech companies in HK is growing which will hopefully even the power side in the future "I know dreamy but possible", so I believe providing startups with tools to help them getting up and running is a good direction specially that most of them under strict or no budget.

Features: 
1- interface to help user generate the desired the test case or playing the whole scenario.
2- Node.js server to listen to and handle incoming requests.
3- ability to generate test reports.
4- Support application level FIX testing.
5- Support Session level FIX testing.

Learning points:
1- Know how FIX works, which is crucial if you want to join fintech/financial companies .
2- Deep understanding of how FIX certification work, also very good to have.
3- Front-end development.
4- Server-side development.
5- We can throw "TDD" (Test Driven Development) into the MIX and learn more about testing and having high code coverage.

Not sure if I forgot anything here, but feel free to add/comment.

**Comment by bgever** [comment]
This resonates a lot. We also hit the issue of testing with the FIX protocol. We use FIXSIM for test trading in our HK based FinTech company, but it isn't flexible enough for automated integration tests. Happy to collaborate on this if it remains purely open-source.

**Comment by ka-sh** [comment]
This is the whole point of doing that, I also want to keep it open source

**Comment by bjdixon** [comment]
I don't have FIX experience but do have node and automated testing experience. I'd be happy to help with this project.

**Comment by ka-sh** [comment]
Okay then, I will start preparing few resources and create the repo.

**Comment by afanasy** [comment]
Any educational links for those who are not familiar with FIX?

**Comment by ka-sh** [comment]
Hey afanasy, please check the repo that I created go to the resources folder and you will find a pdf, very good starting point

**Comment by ka-sh** [comment]
https://github.com/ka-sh/fixsim.js

**Comment by ka-sh** [comment]
Also I am not sure when officially we will start working on that project, but I am more than happy to give like a session or something for FIX and implementations.will help me to refreshing my FIX knowledge.

**Comment by bjdixon** [comment]
A FIX session would probably be a good idea. I'd be interested in that.

**Comment by ka-sh** [comment]
Yeah it actually work both ways, problem is in HK spaces really bad.

**Comment by bgever** [comment]
Maybe @kevinbluer can help to arrange a meeting place at Metta to discuss this?
